junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.platform.launcher.core;
+
+import static java.util.stream.Collectors.joining;
+import static org.apiguardian.api.API.Status.INTERNAL;
+import static org.junit.platform.engine.Filter.composeFilters;
+
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.apiguardian.api.API;
+import org.junit.platform.commons.JUnitException;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.junit.platform.commons.util.BlacklistedExceptions;
+import org.junit.platform.engine.Filter;
+import org.junit.platform.engine.FilterResult;
+import org.junit.platform.engine.TestDescriptor;
+import org.junit.platform.engine.TestEngine;
+import org.junit.platform.engine.UniqueId;
+import org.junit.platform.launcher.EngineDiscoveryResult;
+import org.junit.platform.launcher.LauncherDiscoveryListener;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.launcher.PostDiscoveryFilter;
+
+/**
+ * @since 1.7
+ */
+@API(status = INTERNAL, since = "1.7", consumers = "testkit")
+public class EngineDiscoveryOrchestrator {
+
+	private static final Logger logger = LoggerFactory.getLogger(EngineDiscoveryOrchestrator.class);
+
+	private final EngineDiscoveryResultValidator discoveryResultValidator = new EngineDiscoveryResultValidator();
+	private final Iterable<TestEngine> testEngines;
+
+	public EngineDiscoveryOrchestrator(Iterable<TestEngine> testEngines) {
+		this.testEngines = testEngines;
+	}
+
+	public LauncherDiscoveryResult discover(LauncherDiscoveryRequest request, String phase) {
+		Map<TestEngine, TestDescriptor> testEngineDescriptors = new LinkedHashMap<>();
+
+		for (TestEngine testEngine : this.testEngines) {
+			// @formatter:off
+            boolean engineIsExcluded = request.getEngineFilters().stream()
+                    .map(engineFilter -> engineFilter.apply(testEngine))
+                    .anyMatch(FilterResult::excluded);
+            // @formatter:on
```

Comment:
@marcphilipp Oops, looks like the code was not formatted properly.

Reply:
Thanks, I hate it when IDEA does that. 😬

Reply:
I hate `// @formatter:on/off` commented out directives too. 😁

Reply:
Me too. :)

Reply:
When is 2021?!